index.html

```diff
@@ -929,31 +1106,423 @@ <h2>ConsumedThing Examples</h2>
+              If |type| is `"array"`, run these sub-steps:
+              <ol>
+                <li>
+                  If |source| is not an array, throw {{SyntaxError}} and abort these steps.
+                </li>
+                <li>
+                  Let |length| be the length of |source|.
+                </li>
+                <li>
+                  If |form|'s |minItems| is defined and |length| is less than
+                  that, or if |form|'s |maxItems| is defined and |length| is
+                  more than that, throw {{SyntaxError}} and abort these steps.
+                </li>
+                <li>
+                  For each |item| in |source|, let |itemschema| be |schema|'s
+                  |items| and let |item| be the result of running the
+                  <a>create interaction data</a> steps on |item|, |form| and
+                  |itemschema|.
+                  If this throws, throw {{SyntaxError}} and abort these steps.
+                </li>
+                <li>
+                  Set |data|'s [[\value]] to |source|.
+                </li>
+              </ol>
+            </li>
+            <li>
+              If |type| is `"object"`, run these sub-steps:
+              <ol>
+                <li>
+                  If |source| is not an object, throw {{SyntaxError}} and abort these steps.
+                </li>
+                <li>
+                  If |schema|'s |properties| is not an object,
+                  throw {{SyntaxError}} and abort these steps.
+                </li>
+                <li>
+                  For each property |key| in |source|,
+                  <ol>
+                    <li>Let |value| be the value of |key|.</li>
+                    <li>Let |propschema| be the value of |key| in |properties|.</li>
+                    <li>
+                      Let |value| be the result of running the
+                      <a>create interaction data</a> steps on |value|, |form|
+                      and |propschema|.
+                      If this throws, then throw {{SyntaxError}} and abort these steps.
+                    </li>
+                  </ol>
+                </li>
+                <li>
+                  If |schema|'s |required| is an array, for each |item| in |required|
+                  check if |item| is a property name in |source|. If an |item| is
+                  not found in |source|, throw {{SyntaxError}} and abort these steps.
+                </li>
+                <li>
+                  Set |data|'s [[\value]] to |source|.
+                </li>
+              </ol>
+            </li>
+          </ol>
+        </li>
+        <li>
+          Set |idata|'s |data| to a new {{ReadableStream}} created from |idata|'s
+          [[\value]] <a>internal slot</a> as its
+          <a href="https://streams.spec.whatwg.org/#underlying-source">
+          underlying source</a>.
+        </li>
+        <li>
+          Return |idata|.
+        </li>
+      </ol>
+    </section>
+
```

Comment:
Is this algorithm updated to the latest changes? I miss understanding why we parse source at the creation of `InteractionData`. Shouldn't it be done inside the `value` function?

Reply:
Yes. We have not really discussed writing, but it's a bit different since it takes `any` as value (as aligned with other APIs as well) and one of the values may be a prepared `InteractionData`. But it may be anything else as well, as the algorithm explores. 

If an `InteractionData` is provided, the writing algorithm doesn't use `value()`, it uses its `data`.

Reply:
Now I got it, sorry I misinterpreted the paragraph I thought the algorithm was for creating an InteractionData from the runtime perspective! 

it makes perfect sense now!